actors/builtin/miner/miner_actor.go

```diff
@@ -1221,6 +1226,16 @@ func computeFaultsFromMissingPoSts(st *State, deadlines *Deadlines, sinceDeadlin
 	return
 }
 
+// Check expiry is exactly *the epoch before* the start of a proving period.
+func validateExpiration(st State, expiration abi.ChainEpoch, rt Runtime) {
```

Comment:
Pass `State` as a pointer, not value.

For consistency with other methods, pass `rt` as the first parameter. But better generally would be to return an error from this function and do the abort directly in the actor method, avoding the `rt` parameter. As a policy, it makes these methods callable without needing a runtime (but there are cases where it's pragmatic to ignore).